teps/0090-matrix.md

```diff
@@ -721,6 +720,18 @@ If needed, we can also explore providing more granular controls for maximum numb
 or `Runs` from `Matrices` - either at `PipelineRun`, `Pipeline` or `PipelineTask` levels - later.
 This is an option we can pursue after gathering user feedback - it's out of scope for this TEP.
 
+### Failure Strategies
+
+In failure scenarios, the `TaskRuns` or `Runs` created from a `Matrix` will fail fast. That is, when
+any `TaskRun` or `Run` from a given fanned-out `PipelineTask` fails or is cancelled then the other
+`TaskRuns` or `Runs` from the same `PipelineTask` will be cancelled.
+
+This approach makes it easier to control the execution of the many `TaskRuns` or `Runs` that can be
+created from a `Matrix` (up to 256 for now). Moreover, failing fast is the default behavior of `Matrix`
+in other Continuous Delivery systems - see [GitHub Actions - Handling Failures in Matrix][ghm-failfast].
+
+If needed, we can explore supporting other failure strategies later. 
```

Comment:
This is the right call - in Jenkins Pipeline, we inherited certain assumptions around failure behavior for parallel executions (both in a matrix and not) which resulted in us having to support making fail-fast configurable, and I don't think it was worth doing that in the end. When we're starting from scratch, like here, we have the opportunity to say "this is how things work" and adjust if sufficient demand comes up in the future, rather than prematurely optimizing. So, yeah. 👍